src/test/unit/BLSRegistryCoordinatorWithIndicesUnit.t.sol

```diff
@@ -0,0 +1,649 @@
+
+        for (uint i = 0; i < numOperators; i++) {
+            BN254.G1Point memory pubKey = BN254.hashToG1(keccak256(abi.encodePacked(pseudoRandomNumber, i)));
+            address operator = _incrementAddress(defaultOperator, i);
+            
+            _registerOperatorWithCoordinator(operator, quorumBitmap, pubKey);
+        }
+
+        address operatorToRegister = _incrementAddress(defaultOperator, numOperators);
+        BN254.G1Point memory operatorToRegisterPubKey = BN254.hashToG1(keccak256(abi.encodePacked(pseudoRandomNumber, numOperators)));
+    
+        pubkeyCompendium.setBLSPublicKey(operatorToRegister, operatorToRegisterPubKey);
+
+        stakeRegistry.setOperatorWeight(defaultQuorumNumber, operatorToRegister, defaultStake);
+
+        cheats.prank(operatorToRegister);
+        cheats.expectRevert("BLSIndexRegistryCoordinator._registerOperatorWithCoordinatorAndNoOverfilledQuorums: quorum is overfilled");
+        registryCoordinator.registerOperatorWithCoordinator(quorumNumbers, operatorToRegisterPubKey, defaultSocket);
+    }
+
+    function testDeregisterOperatorWithCoordinator_NotRegistered_Reverts() public {
+        bytes memory quorumNumbers = new bytes(1);
+        quorumNumbers[0] = bytes1(defaultQuorumNumber);
+
+        cheats.expectRevert("BLSIndexRegistryCoordinator._deregisterOperatorWithCoordinator: operator is not registered");
+        cheats.prank(defaultOperator);
+        registryCoordinator.deregisterOperatorWithCoordinator(quorumNumbers, defaultPubKey, new bytes32[](0));
+    }
+
+    function testDeregisterOperatorWithCoordinator_IncorrectPubkey_Reverts() public {
+        bytes memory quorumNumbers = new bytes(1);
+        quorumNumbers[0] = bytes1(defaultQuorumNumber);
+        uint256 quorumBitmap = BitmapUtils.orderedBytesArrayToBitmap(quorumNumbers);
+
+        _registerOperatorWithCoordinator(defaultOperator, quorumBitmap, defaultPubKey);
+
+        BN254.G1Point memory incorrectPubKey = BN254.hashToG1(bytes32(uint256(123)));
+
+        cheats.expectRevert("BLSIndexRegistryCoordinator._deregisterOperatorWithCoordinator: operatorId does not match pubkey hash");
+        cheats.prank(defaultOperator);
+        registryCoordinator.deregisterOperatorWithCoordinator(quorumNumbers, incorrectPubKey, new bytes32[](0));
+    }
+
+    function testDeregisterOperatorWithCoordinator_InvalidQuorums_Reverts() public {
+        bytes memory quorumNumbers = new bytes(1);
+        quorumNumbers[0] = bytes1(defaultQuorumNumber);
+        uint256 quorumBitmap = BitmapUtils.orderedBytesArrayToBitmap(quorumNumbers);
+
+        _registerOperatorWithCoordinator(defaultOperator, quorumBitmap, defaultPubKey);
+
+        quorumNumbers = new bytes(2);
+        quorumNumbers[0] = bytes1(defaultQuorumNumber);
+        quorumNumbers[1] = bytes1(uint8(192));
+
+        cheats.expectRevert("BLSIndexRegistryCoordinator._deregisterOperatorWithCoordinator: quorumsToRemoveBitmap cant have more than 192 set bits");
```

Comment:
this revert message seems a tad...cryptic? maybe it should be something like 'quorum bitmap out of max bitmap size / range'? -- also maybe we should have a tiny internal function just for this, that we reuse the few places we do the check. any thoughts?

Reply:
40533cf7 chose to not do the internal function, what do u anticipate being recorded?